Remediate BRR-03C

cliffhall opened the pull request:
* In BosonRouter.sol,
  - requestCreateOrder and checkLimits methods
    - Verified that all callers are passing calldata that would otherwise be copied needlessly to memory
    - _metadata argument
       - Changed location from memory to calldata
       - Verified compilation and unit tests all succeed.
* This fixes #180

cliffhall committed:
Remediate BRR-03C

* In BosonRouter.sol,
  - requestCreateOrder method
    - Verified that all callers are passing calldata that would otherwise be copied needlessly to memory
    - _metadata argument
       - Changed location from memory to calldata
       - Verified compilation and unit tests all succeed.

zajck reviewed (changes requested):
Funcion `checkLimits` should be updated too.

hswopeams reviewed (commented):
I agree with @zajck that `checkLimits` needs to be changed too.

cliffhall committed:
Merge branch 'main' into remediate-BRR-03C

cliffhall committed:
Remediate BRR-03C

* In BosonRouter.sol,
  - in checkLimits method
    - changed location of _metadata argument from memory to calldata
